protocols/data_processor.py

```
@@ -4,101 +4,105 @@
 import os
 from datetime import datetime
 
+
 def task():
```

Ruff (0.11.9)

`task` is too complex (13 > 10)

(C901)